src/server/transaction.cc

```diff
@@ -116,6 +116,40 @@ uint32_t Transaction::PhasedBarrier::DEBUG_Count() const {
   return count_.load(memory_order_relaxed);
 }
 
+bool Transaction::SingleClaimBarrier::IsClaimed() const {
+  return claimed_.load(memory_order_relaxed);
+}
+
+bool Transaction::SingleClaimBarrier::TryClaim() {
+  return !claimed_.exchange(true, memory_order_relaxed);  // false means first means success
```

Comment:
I am trying to understand why relaxed is enough here. Do not see any obvious reasons why not and it makes me nervous. Usually, it should not be like this.

Reply:
1. It was relaxed to begin with 😆 
2. Currently code touches only shard local state or expire/cancel are placed on the coordinator thread
3. If we acquire, what is our corresponding release pair? We had an acuiqre/release for the hop that prepared suspension, since then we did no writes to our local state, so there is noting to sync
4. I can make it acquire because though I assume it might have real impact on the conclusion writes

Will think about it